src/libopenrave/kinbody.cpp

```diff
@@ -6154,7 +6163,7 @@ UpdateFromInfoResult KinBody::UpdateFromKinBodyInfo(const KinBodyInfo& info)
             // if any link has its transform field set, we need to set zero configuration before comparison
             if( (*itLinkInfo)->IsModifiedField(KinBody::LinkInfo::LIF_Transform) ) {
                 stateSaver.reset(new KinBody::KinBodyStateSaver(shared_kinbody(), Save_LinkTransformation));
-                SetTransform(Transform());
+                SetTransform(_baseLinkRelativeTransform); // so that base link is at exactly _baseLinkRelativeTransform
```

Comment:
why not keep it as `Transform()`?

Reply:
Originally, I was trying to preserve `GetLinks()[0]->GetTransform() == GetTransform()`. So I needed to put the body temporarily at `base link in body frame` transform, such that the `GetLinks()[0]->GetTransform()` would then be equal to the original `base link in body frame` transform when things are not changed.